htdocs/lib2/translationHandler.class.php

```diff
@@ -238,7 +238,7 @@ public function substitueVariables(&$variables, $lang, $str)
 
         // replace variables in string
         if (mb_ereg_search_init($langstr)) {
-            while (false != $vars = mb_ereg_search_regs("%[^%]*%")) {
+            while ($vars = mb_ereg_search_regs('%[^%]*%') != false) {
```

Comment:
**This introduces a bug.** `$vars` should be set to the result of the regex search, but now it is set to the result of comparing the regex result with `false`.

All the rest of this refactoring should be carefully examined for similiar constructs which may have been broken.